build/mongodl.py

```diff
@@ -84,12 +89,15 @@ def infer_target():
 
 
 def _infer_target_os_rel():
-    content = Path('/etc/os-release').read_text()
+    with Path('/etc/os-release').open('r', encoding='utf-8') as f:
+        content = f.read()
     id_re = re.compile(r'\bID=("?)(.*)\1')
     mat = id_re.search(content)
     assert mat, 'Unable to detect ID from [/etc/os-release] content:\n{}'.format(
         content)
     os_id = mat.group(2)
+    if os_id == 'arch':
+        return 'rhel80'
```

Comment:
Why is `'arch'` special-cased here instead of using `DISTRO_ID_MAP` and `DISTRO_ID_TO_TARGET`? Git blame says Arch is a special case but does not explain why (would be nice if rationale is also documented as a comment).